cmd/wallet-web/src/pages/mixins/common/helper.js

```diff
@@ -89,6 +89,52 @@ export const isVPType = (type) => toLower(type) == 'verifiablepresentation';
 export const getCredentialType = (types) =>
   types.filter((type) => type != 'VerifiableCredential')[0];
 
+// function to replace json path from the json objects
+export function getCredentialDisplayData(vc, manifest) {
+  // title
+  const titleFound = findByPath(vc, manifest.title.path);
+  const title = titleFound ? titleFound : manifest.title.fallback;
+
+  // icon
+  const icon = manifest.icon;
+
+  // find properties
+  const _readProperty = (property) => {
+    let pKey = property[0];
+    let vals = findByPath(vc, property[1].path);
+    const value = vals ? vals : '';
+    const label = property.label;
+    const type = property.type;
+    const format = property.format;
+    return {
+      [pKey]: {
+        label,
+        value,
+        type,
+        format,
+      },
+    };
+  };
+
+  const properties = Object.entries(manifest.properties).map(_readProperty);
+  return {
+    title,
+    icon,
+    properties,
+  };
+}
+// function to replace json path from the json objects
+function findByPath(vc, ...paths) {
+  let pathValue;
+  paths.forEach((path) => {
+    path.forEach((nestedPath) => {
+      pathValue = jp.query(vc, nestedPath);
+    });
+  });
+  // returns the first value which is path[0].
+  return pathValue.shift();
```

Comment:
Why do you need to iterate two times and call a jp.query function on each iteration? This is very expensive and does not seem right

Reply:
updated to single for loop, thanks

Reply:
why it is nested array?

Reply:
its not it way too simplified in the recent option